canvas_grab/snapshot/canvas_module_snapshot.py

```diff
@@ -1,74 +1,106 @@
+            with_link (bool, optional): If true, pages will be included in snapshot. Defaults to False.
+        """
         self.course = course
         self.snapshot = {}
         self.with_link = with_link
 
     def add_to_snapshot(self, key, value):
+        """Add a key-value pair into snapshot. If duplicated, this function will report error and ignore the pair.
+
+        Args:
+            key (str): key or path of the object
+            value (any): content of the object
+        """
         if key in self.snapshot:
             print(colored(
                 f'  Duplicated file found: {key}, please download it using web browser.', 'yellow'))
             return
         self.snapshot[key] = value
 
     def take_snapshot(self):
+        """Take a snapshot
+
+         Returns:
+             dict: snapshot of Canvas in `SnapshotFile` or `SnapshotLink` type.
+         """
         course = self.course
-        request_batcher = RequestBatcher(course)
-        accessed_files = []
-
-        for _, module in (request_batcher.get_modules() or {}).items():
-            # replace invalid characters in name
-            name = re.sub(file_regex, "_", module.name)
-            # consolidate spaces
-            name = " ".join(name.split())
-
-            # get module index
-            idx = str(module.position)
-
-            # folder format
-            module_name = f'{idx} {name}'
-
-            module_item_count = module.items_count
-            print(
-                f'  Module {colored(module_name, "cyan")} ({module_item_count} items)')
-
-            for item in module.get_module_items():
-                if item.type == 'File':
-                    file_id = item.content_id
-                    snapshot_file = from_canvas_file(
-                        request_batcher.get_file(file_id))
-                    accessed_files.append(file_id)
-                    filename = f'{module_name}/{snapshot_file.name}'
-                    self.add_to_snapshot(filename, snapshot_file)
-                if self.with_link:
-                    if item.type == 'ExternalUrl' or item.type == 'Page':
-                        key = f'{module_name}/{item.title}.html'
-                        value = SnapshotLink(
-                            item.title, item.html_url, module_name)
-                        self.add_to_snapshot(key, value)
-
-        files = request_batcher.get_files()
-        if files:
-            unmoduled_files = 0
-            for file_id, file in files.items():
-                if file_id not in accessed_files:
-                    snapshot_file = from_canvas_file(file)
-                    filename = f'unmoduled/{snapshot_file.name}'
-                    self.add_to_snapshot(filename, snapshot_file)
-                    unmoduled_files += 1
-            print(
-                f'  {colored("Unmoduled files", "cyan")} ({unmoduled_files} items)')
+         request_batcher = RequestBatcher(course)
```

Comment:
Wrong indentation here @skyzh

Reply:
Fixed in https://github.com/skyzh/canvas_grab/pull/134